### ISSUE TITLE
integrated with CREPA in simpletuner

### ISSUE THREAD

**Issue by bghira** [vita-epfl/LayerSync#1]
hello! thank you for the publication and availability of source code to make it much less ambiguous.

i've integrated layersync into simpletuner by adding into each transformer model, a hidden state buffer which can return full values. this will be helpful for other experiments, so it was a good idea in general.

the core layersync implementation is plumbed into flux.1+2, pixart, SD3, SDXL, Sana, Wan, Qwen, Chroma, ACE-Step (audio), HiDream, cosmos, LongCat, Z-image, kandinsky5 image, and Auraflow.

the key integration point with CREPA is to replace the DINO features so we no longer require the external encoder. this works for LTX Video, Sana Video, Wan, Hunyuan Video 1.5, and Kandinsky5 Video.

if you have time, i have a few questions that came up during development:

are there any per-family default block pairs you prefer for layersync beyond the 8→16 guess on 24-layer dits?
is 0.2 still your preferred weight when batching with other aux losses?
any guidance on handling 3d video tokens (b, t, p, d) vs 2d (b, s, d)—should we treat temporal tokens differently?
any concerns with the 1-based-then-0-based index resolution we’re doing to match your paper/rep0?

my pull request is here: https://github.com/bghira/SimpleTuner/pull/2161

it's easily configured by the webUI:

<img width="1528" height="315" alt="Image" src="https://github.com/user-attachments/assets/4ac39c5b-a325-4dfd-b832-8ed8f316c282" />

**Comment by Yasaman-Haghighi** [comment]
Thank you for your interest in our work!

Regarding your question about layer matching, we conducted a detailed ablation study to identify which layers yield the best results. You can find these results in the attached image.

In summary, we categorize the transformer blocks into three distinct functional groups: Initial Layers, focus primarily on local features. Middle Layers, focus on global features and Final Layers act as a decoder. (see [Raghu, Maithra, et al. "Do vision transformers see like convolutional neural networks?." Advances in neural information processing systems 34 (2021): 12116-12128.] for more details).

As shown by the ablation study, the performance gain is optimized when matching blocks within the middle group.

Implementation Details:
Weighting: Since LayerSync functions as a regularizer, it is important to ensure the weight is balanced so it doesn't overpower the primary diffusion loss.

Video vs. Image: We did not observe a significant difference between video and image processing. This is because our alignment occurs between individual tokens and their corresponding counterparts in other layers. To implement this, you can simply reshape the video tensor from $(B, T, P, D)$ to $(B, S, D)$ (where $S = T \times P$) and synchronize the representations.

Please don't hesitate to reach out if you have any further questions!

<img width="505" height="663" alt="Image" src="https://github.com/user-attachments/assets/7f8a846d-c2cf-4370-bc3d-b3439bfd9278" />

**Comment by Yasaman-Haghighi** [comment]
Hello! We've added an example demonstrating how to integrate LayerSync when fine-tuning with LoRA: [Finetuning-LoRA](https://github.com/vita-epfl/LayerSync/tree/main/Finetuning-LoRA)
We aligned layers 8–16 with a weight of 0.02.
